submissions/AndriiZahumnyi/a-tiny-js-world-pre-oop/index.js

```diff
@@ -0,0 +1,101 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/AndriiZahumnyi/a-tiny-JS-world
+   Web app: https://andriizahumnyi.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+
+
+const man = {
+   species: 'human',
+   name: 'John',
+   gender: 'male',
+   legs: 2,
+   hands: 2,
+   saying: 'Hi! How are you?',
+   friendlyTo: ['Toby', 'Tereza', 'Tom']
+};
+
+const woman = {
+   species: 'human',
+   name: 'Tereza',
+   gender: 'female',
+   legs: 2,
+   hands: 2,
+   saying: 'I`m fine, thanks!',
+   friendlyTo: ['Tom']
+};
+
+const dog = {
+   species: 'dog',
+   name: 'Toby',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   tail: 1,
+   saying: 'woof-woof!',
+   friendlyTo: ['John', 'Tereza']
+};
+
+const cat = {
+   species: 'cat',
+   name: 'Tom',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   tail: 1,
+   saying: 'meow-meow!',
+   friendlyTo: ['Tereza']
+};
+
+const catWoman = {
+   species: 'human',
+   name: 'Kitty',
+   gender: 'female',
+   legs: 2,
+   hands: 2,
+   tail: 1,
+   saying: cat.saying
+};
+
+const world = [man, woman, catWoman, dog, cat];
+
+const listOfKeys = ['species', 'name', 'gender', 'legs', 'hands', 'tail', 'saying', 'friendlyTo'];
+
+makingString = obj => {
+   let string = '';
+   for (let i = 0; i < listOfKeys.length; i++) {
+      if (i + 1 === listOfKeys.length) {
+         string += (obj[listOfKeys[i]] + ".");
+         return string;
+      } else {
+         string += (obj[listOfKeys[i]] + "; ");
+      }
+   }
+};
```

Comment:
Employ `Array.map` to convert a list of strings into values of the properties. Then join with a delimiter (`"; "` I guess) and add final `"."`
This will make code cleaner, shorter and easier to read.

Reply:
You meant "to convert a list of the properties into strings" I think.

I`ve use method map, as you advised, but it doesn`t print undefined properties if property is absent and instead print extra ";" one more time. Is it better?

Reply:
@AndriiZahumnyi 
Consider the following.
`[undefined, 7, undefined, 9, 11, 0].filter(v => v !== undefined)` results in `[7, 9, 11, 0]`